internal/publish/model/exposure_model.go

```diff
@@ -17,17 +17,78 @@ package model
 
 import (
 	"context"
+	"encoding/base64"
 	"fmt"
+	"math"
 	"sort"
 	"strings"
 	"time"
 
 	"github.com/google/exposure-notifications-server/internal/logging"
+	"github.com/google/exposure-notifications-server/internal/verification"
 	"github.com/google/exposure-notifications-server/pkg/base64util"
 
 	verifyapi "github.com/google/exposure-notifications-server/pkg/api/v1alpha1"
 )
 
+// Exposure represents the record as stored in the database
+// TODO(mikehelmick) - refactor this so that there is a public
+// Exposure struct that doesn't have public fields and an
+// internal struct that does. Separate out the database model
+// from direct access.
+// Mark records as writable/nowritable - is exposure key encrypted.
+type Exposure struct {
+	ExposureKey      []byte
+	TransmissionRisk int
+	AppPackageName   string
+	Regions          []string
+	IntervalNumber   int32
+	IntervalCount    int32
+	CreatedAt        time.Time
+	LocalProvenance  bool
+	FederationSyncID int64
+
+	// These fileds are nullable to maintain backwards compatibility with
+	// older versions that predate their existence.
+	HealthAuthorityID            *int64
+	ReportType                   string
+	DaysSinceSymptomOnset        *int32
+	RevisedReportType            *string
+	RevisedAt                    *time.Time
+	RevisedDaysSinceSymptomOnset *int
+}
+
+// HasDaysSinceSymptomOnset returns true if the this key has the days since
+// symptom onset field is et.
+func (e *Exposure) HasDaysSinceSymptomOnset() bool {
+	return e.DaysSinceSymptomOnset != nil
+}
+
+// SetDaysSinceSymptomOnset sets the days since sympton onset field, possibly
+// allocating a new pointer.
+func (e *Exposure) SetDaysSinceSymptomOnset(d int32) {
+	if e.DaysSinceSymptomOnset == nil {
+		e.DaysSinceSymptomOnset = new(int32)
+	}
+	*e.DaysSinceSymptomOnset = d
+}
+
+func (e *Exposure) HasHealthAuthorityID() bool {
+	return e.HealthAuthorityID != nil
+}
+
+func (e *Exposure) SetHealthAuthorityID(haID int64) {
+	if e.HealthAuthorityID == nil {
+		e.HealthAuthorityID = new(int64)
+	}
+	*e.HealthAuthorityID = haID
+}
+
+// ExposureKeyBase64 returns the ExposuerKey property base64 encoded.
+func (e *Exposure) ExposureKeyBase64() string {
+	return base64.StdEncoding.EncodeToString(e.ExposureKey)
```

Comment:
STD or RAW? Padded or non?

Reply:
exposure keys have always been StdEncoding in our use in the database.